shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1799,11 +1857,74 @@ - (void)scrollEvent:(UIPanGestureRecognizer*)recognizer API_AVAILABLE(ios(13.4))
     _mouseState.last_translation = CGPointZero;
   }
 
+  if (_mouseState.flutter_state_is_added) {
+    packet->SetPointerData(/*index=*/0, pointer_data);
+  } else {
+    flutter::PointerData add_pointer_data = pointer_data;
+    add_pointer_data.signal_kind = flutter::PointerData::SignalKind::kNone;
+    add_pointer_data.change = flutter::PointerData::Change::kAdd;
+    _mouseState.flutter_state_is_added = true;
+    packet->SetPointerData(/*index=*/0, add_pointer_data);
+    packet->SetPointerData(/*index=*/1, pointer_data);
+  }
+
+  [_engine.get() dispatchPointerDataPacket:std::move(packet)];
+}
+
+- (void)panEvent:(UIPanGestureRecognizer*)recognizer API_AVAILABLE(ios(13.4)) {
+  CGPoint translation = [recognizer translationInView:self.view];
+  const CGFloat scale = [UIScreen mainScreen].scale;
+
+  translation.x *= scale;
+  translation.y *= scale;
+
+  flutter::PointerData pointer_data = [self generatePointerDataForMouse];
```

Comment:
I think pan is for track pad not mouse, right? The method `generatePointerDataForMouse` might need some renaming to avoid confusion.

Reply:
Since it's for iPad, it's already the minority to have a mouse or trackpad at all. Do you think it's better to use generatePointerDataForCursor ?

Reply:
Yeah for cursor is better, mouse seems pretty specific to the device that used to control the cursor. Which is not true for "pan" gesture. "Pan" is track pad only right?

Reply:
Pan gesture could come from Apple Magic Mouse